api/models.js

```diff
@@ -285,6 +285,18 @@ var publicationSchema = mongoose.Schema({
     create_date: { type: Date, default: Date.now },
 
     removed: { type: Boolean, default: false }, //only admin can remove publication for now (so that doi won't break)
+    mag: {
+        papers: [{  
```

Comment:
Is this the same schema for project.mag.papers? If so, please create a new sub-schema and refactor this